### PR TITLE
Added custom headers to api functions

### PR DESCRIPTION
**Pull request by thecodingtree**

### CODE REVIEW

File: src/sagas/api.js [hunk 1]
```diff
@@ -4,74 +4,115 @@ import { call, select } from 'redux-saga/effects';
...
+import * as _ from 'lodash';
+
+const parseOptions = (options) => {
+  let data = null;
+  let customHeaders = undefined;
+
+  if (_.has(options, 'data')) {
+    data = _.get(options, 'data');
+
+    if (_.has(options, 'headers')) {
```
Comment (KellyInnovation) [inline src/sagas/api.js:16]: Do we only want this if there is data?  What if we want custom headers on a `get` request that doesn't have data?

Comment (chris-allen) [reply src/sagas/api.js:16]: +1

File: README.md [hunk 2]
```diff
@@ -205,3 +238,7 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
+
+```
```
[the remark this answers is not in the document]
Comment (thecodingtree) [reply README.md]: removed

File: src/sagas/api.js [hunk 3]
```diff
@@ -4,79 +4,132 @@ import { call, select } from 'redux-saga/effects';
...
 
-export function* getHeaders (authenticationRequired) {
+import * as _ from 'lodash';
+
+const parseOptions = (options) => {
+  let data = null;
+  let customHeaders = undefined;
+  let authenticationRequired = true;
+
+  let hasData = _.has(options, 'data');
```
Comment (KellyInnovation) [inline src/sagas/api.js:14]: It looks like these should be `const` instead as they don't change.  Alternatively, 
`
if (_.has(options, 'data')) {
  data = _.get(options, 'data');
}

if (_.has(options, 'headers')) {
  customHeaders = _.get(options, 'headers');
}

if (_.has(options, 'authenticationRequired')) {
  authenticationRequired = _.get(options, 'authenticationRequired');
}
`

**Pull request merged**